src/driver/mpas_subdriver.F

```diff
@@ -109,38 +111,68 @@ end subroutine xml_stream_get_attributes
       readNamelistArg = .false.
       readStreamsArg = .false.
 
-      nArgs = command_argument_count()
-      iArg = 1
-      do while (iArg < nArgs)
-         call get_command_argument(iArg, argument)
-         if (len_trim(argument) == 0) exit
-
-         if ( trim(argument) == '-n' ) then
-            iArg = iArg + 1
+      ! If provided, error check the namelistFileParam and copy it to namelistFile to override default
+      if (present(namelistFileParam)) then
+         if (len_trim(namelistFileParam) <= 0) then
+            write(0,'(A)') 'WARNING: mpas_init argument ''namelistFileParam'' has 0 or negative length and will be ignored'
+         else if (len_trim(namelistFileParam) > len(namelistFile)) then
+            write(0,'(A,I5,A,I5,A)') 'CRITICAL ERROR: mpas_init argument ''namelistFileParam'' has length ', len_trim(namelistFileParam),&
+                                     ', but the maximum allowed is ', len(namelistFile), ' characters'
+            stop
+         else
             readNamelistArg = .true.
-            call get_command_argument(iArg, namelistFile)
-            if ( len_trim(namelistFile) == 0 ) then
-                write(0,*) 'ERROR: The -n argument requires a namelist file argument.'
-                stop
-            else if ( trim(namelistFile) == '-s' ) then
-                write(0,*) 'ERROR: The -n argument requires a namelist file argument.'
-                stop
-            end if
-         else if ( trim(argument) == '-s' ) then
-            iArg = iArg + 1
+            namelistFile = trim(namelistFileParam)
+         endif
```

Comment:
Another minor stylistic issue: for the same reason we write `else if` with a space, I think `end if` is preferable to `endif`. Also, one can `end` many things in Fortran: `do`, `subroutine`, `program`, etc., so why not `if` as well?

Reply:
Gotcha, that can be easily fixed.